service/history/failover/coordinator.go

```diff
@@ -255,6 +259,14 @@ func (c *coordinatorImpl) handleFailoverMarkers(
 		record.shards[shardID] = struct{}{}
 	}
 
+	domainEntry, err := c.domainCache.GetDomainByID(domainID)
```

Comment:
nit: if all you need is the domainName, consider using the `GetDomainName()` method which can avoid a deep copy of domainEntry.

Reply:
Good call. will update

Reply:
Let we create an issue for this actually, to use `GetDomainName` whenever possible. The deep copy of domain entry allocated many objects and created quite some GC pressure when I was testing in staging2.